scala/scala-impl/src/org/jetbrains/sbt/project/structure/MillProjectDumper.scala

```diff
@@ -0,0 +1,29 @@
+package org.jetbrains.sbt.project.structure
+
+import java.io.File
+import java.util.concurrent.atomic.AtomicBoolean
+import scala.concurrent.duration._
+import scala.collection.JavaConverters._
+
+object MillProjectDumper {
+  private val MillProcessCheckTimeout = 100.millis
+
+  def setupBsp(baseDir: File): MillProjectDumper = {
+    val processBuilder = new ProcessBuilder(Seq("./mill", "-i", "mill.contrib.BSP/install").asJava)
```

Comment:
is the mill script available in every mill project, and will it work like this on Windows?

Reply:
No, ideally we would not need to rely on this and would have an equivalent of `sbt-launcher.jar` but I didn't have time to figure out how this whole thing works.
The `canImport` checks for that script but it's true that on windows that would be named `mill.bat` I think.

Reply:
I think it's fine if we can't cover all setups for now as long as it doesn't break in situations where assumptions don't hold